core-bundle/src/Filesystem/VirtualFilesystem.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Filesystem;
+
+use League\Flysystem\DirectoryListing;
+use League\Flysystem\FilesystemOperator;
+use League\Flysystem\StorageAttributes;
+use League\Flysystem\UnableToResolveFilesystemMount;
+use Symfony\Component\Filesystem\Path;
+
+/**
+ * A virtual filesystem is Contao's version of a MountManager. You can operate
+ * on it like on any other FilesystemOperator but under the hood calls are
+ * redirected to other FilesystemOperator instances depending on the path
+ * prefix. A more specific prefix always wins over a more general one.
+ *
+ * Example:
+ *   If operator A is mounted to 'files' and operator B to 'files/media',
+ *   calling read('files/media/foo') will issue a call to B->read('foo')
+ *   while calling fileExists('files/bar/baz') will issue a call to
+ *   A->fileExists('bar/baz') instead.
+ *
+ * todo: handle and rethrow exceptions
+ */
+class VirtualFilesystem implements FilesystemOperator
```

Comment:
Note to myself: both the `DbafsFilesystem` and the `VirtualFilesystem` should probably implement an interface that includes metadata support. Maybe `DbafsFilesystemOperator extends FilesystemOperator`?

Reply:
implemented in e21aa8064f352740870ee50415f3d3d37121d29a